plex-service/src/main/java/com/cleverpine/plex/controller/MovieController.java

```diff
@@ -30,44 +30,27 @@ public ResponseEntity<String> apiMoviesTestGet() {
 
     @Override
     @ViravaSecured(resource = Resources.MOVIES, scope = ScopeType.READ)
-    public ResponseEntity<MoviesListResponse> apiMoviesGet(Integer page, Integer size) {
-        try {
-            List<MovieDto> movieList = movieService.getMovieList(page, size);
-
-            MoviesListResponse response = new MoviesListResponse();
-            movieList.stream()
-                    .map(movieMapper::movieDtoToMovieListItem)
-                    .forEach(response::addDataItem);
-
-            return ResponseEntity.ok(response);
-        } catch (Exception e) {
-            return ResponseEntity.notFound().build();
-        }
+    public ResponseEntity<MoviesListResponse> getPaginatedMovieList(Integer page, Integer size) {
+        List<MovieDto> movieList = movieService.getMovieList(page, size);
+        MoviesListResponse response = new MoviesListResponse();
+        response.setData(movieMapper.movieDtoToMovieListItem(movieList));
```

Comment:
Generally it's better to not map the response data inside the controller class. If The getMovieList method is not used elsewhere, refactor it to directly return MovieListItem instead of mapping the data once in the service and once in the controller